crates/sui-indexer/src/handlers/checkpoint_handler.rs

```diff
@@ -171,11 +169,17 @@ where
                     seq, e
                 ))
             });
-        // this happens very often b/c checkpoint indexing is faster than checkpoint
-        // generation. Ideally we will want to differentiate between a real error and
-        // a checkpoint not generated yet.
+        let mut fn_checkpoint_guard = self
+            .metrics
+            .fullnode_checkpoint_download_latency
+            .start_timer();
         while checkpoint.is_err() {
+            // sleep for 1 second and retry if latest checkpoint is not available yet
             tokio::time::sleep(std::time::Duration::from_secs(1)).await;
+            fn_checkpoint_guard = self
```

Comment:
In the summary you mentioned `the sleep time is 1 second, which is too long as checkpoint is generated every a couple of seconds`, does this PR change this behavior?

Reply:
synced separately on slack